moveit_ros/visualization/motion_planning_rviz_plugin/src/motion_planning_frame_states.cpp

```diff
@@ -249,4 +259,169 @@ void MotionPlanningFrame::clearStatesButtonClicked()
   populateRobotStatesList();
 }
 
+void MotionPlanningFrame::planAllStatesButtonClicked()
+{
+#if 0
+  planning_display_->addBackgroundJob(boost::bind(&MotionPlanningFrame::planAllStatesCartesianExec, this),
+                                      "plan all cartesian states");
+#else
+  planning_display_->spawnBackgroundJob(boost::bind(&MotionPlanningFrame::planAllStatesJointSpaceExec, this));
+#endif
+}
+
+void MotionPlanningFrame::planAndExecuteAllStatesButtonClicked()
+{
+#if 0
+  planning_display_->addBackgroundJob(boost::bind(&MotionPlanningFrame::planAndExecuteAllStatesCartesianExec, this),
+                                      "plan and execute all cartesian states");
+#else
+  planning_display_->spawnBackgroundJob(boost::bind(&MotionPlanningFrame::planAndExecuteAllStatesJointSpaceExec, this));
+#endif
+}
+
+void MotionPlanningFrame::planAllStatesCartesianTrajectory()
```

Comment:
I'm not convinced this planning logic belongs in the Qt-based motion_planning_frame_states widget... can we move it to a helper function?